src/html5/parser.rs

```diff
@@ -3905,4 +3894,56 @@ mod test {
         binding.set_node_named_id(NodeId(4), "otherid");
         assert!(binding.get_node_by_named_id("myid").is_none());
     }
+
+    #[test]
+    fn tests2_dat_720() {
+        let document = parse("<!DOCTYPE html>X</html>X");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "XX"
+            "#,
+        );
+    }
```

Comment:
Test `tests2_dat_720` was passing before this PR, and I just used it to debug the tree display format.